packages/css/src/plugins/at-rules/parsers.ts

```diff
@@ -0,0 +1,231 @@
+import type {
+  BasicMatchInfo,
+  ComparisonOperator,
+  LengthInfo,
+  LengthUnit,
+  OperatorInfo,
+  ParsedAtRule,
+  Property,
+  PropertyInfo,
+} from './types';
+
+function isPropertyValid(property: string): property is Property {
+  return ['width', 'height', 'device-width', 'device-height'].includes(property);
+}
+
+function isComparisonOperatorValid(
+  operator: string
+): operator is OperatorInfo['comparisonOperator'] {
+  return ['<=', '=', '>=', '<', '>'].includes(operator);
+}
+
+const convertMinMaxMediaQuery = (
+  match: RegExpMatchArray
+): (PropertyInfo & OperatorInfo) | undefined => {
+  const property = match.groups?.property;
+  switch (property) {
+    case 'min-width':
+    case 'min-device-width':
+      // min-width: XYZ
+      // is the same as
+      // width >= XYZ
+      return {
+        property: property === 'min-width' ? 'width' : 'device-width',
+        comparisonOperator: '>=',
+      };
+    case 'max-width':
+    case 'max-device-width':
+      // max-width: XYZ
+      // is the same as
+      // width <= XYZ
+      return {
+        property: property === 'max-width' ? 'width' : 'device-width',
+        comparisonOperator: '<=',
+      };
+    case 'min-height':
+    case 'min-device-height':
+      return {
+        property: property === 'min-height' ? 'height' : 'device-height',
+        comparisonOperator: '>=',
+      };
+    case 'max-height':
+    case 'max-device-height':
+      return {
+        property: property === 'max-height' ? 'height' : 'device-height',
+        comparisonOperator: '<=',
+      };
+    case undefined:
+      return;
+    default:
+      throw new SyntaxError(`Unexpected property '${property}' found when sorting media queries`);
+  }
+};
+
+/**
+ * This runs when we expect
+ *
+ *     <width|height> <comparisonOperator> <length>
+ *
+ * but we actually got
+ *
+ *     <length> <comparisonOperator> <width|height>
+ *
+ * So we need to normalise the comparison operator by reversing it.
+ *
+ * @param operator Comparison operator.
+ * @returns The reversed comparison operator.
+ */
+const reverseComparisonOperator = (operator: ComparisonOperator): ComparisonOperator => {
+  switch (operator) {
+    case '<':
+      return '>';
+    case '>':
+      return '<';
+    case '<=':
+      return '>=';
+    case '>=':
+      return '<=';
+    case '=':
+      return '=';
+  }
+};
+
+const getProperty = (match: RegExpMatchArray): PropertyInfo | undefined => {
+  const property = match.groups?.property;
+
+  if (property === undefined) {
+    return;
+  }
+
+  if (isPropertyValid(property)) {
+    return {
+      property,
+    };
+  }
+
+  throw new SyntaxError(`Unexpected property '${property}' found when sorting media queries.`);
```

Comment:
Should this category of invariant (here and below) be mentioned in the changeset?

Reply:
hmm

assuming i didn't screw up somewhere, the only situation where this error would be triggered if someone used `max`/`min` in a place where it wasn't valid CSS syntax, such as `@media (max-width < 500px)`. I suppose we would go from silently accepting invalid CSS to now giving a build-time error.

the comparison operator error would only be triggered if there is a bug in the code 😅

i suppose we *could* add this to the changeset, but not sure how much value this would provide

Reply:
We would probably know real quick in our primary codebases, right?  Could be worth a major if it has a semi-reasonable chance to throw odd errors here for existing setups, but up to you.

Reply:
ah minor is what we use as a de-facto "major release" because we still don't have a v1.0 release yet

we should definitely release a v1.0 eventually but haven't had the chance to do it yet